### PR TITLE
Add handling for `DeviceCloseRequest`

### PR DESCRIPTION
**Pull request by ibeckermayer**

### CODE REVIEW

**Review by probakowski — APPROVED** [review]

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs [hunk 1]
```diff
@@ -480,102 +514,126 @@ impl FilesystemBackend {
...
+            return Ok(());
         }
         let file = file.unwrap(); // safe because of the if statement above
         let device_io_response =
             efs::DeviceIoResponse::new(rdp_req.device_io_request.clone(), NtStatus::SUCCESS);
 
         // We support all the FsInformationClasses that FreeRDP does here
         // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L482
-        match rdp_req.file_info_class_lvl {
+        let rdp_resp = match rdp_req.file_info_class_lvl {
```
Comment (ibeckermayer) [inline lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:544]: Moved from https://github.com/gravitational/teleport/pull/34082#discussion_r1380604200

@probakowski:
> Can we extract branches below to separate methods? It's hard to follow

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs [hunk 2]
```diff
@@ -480,102 +514,126 @@ impl FilesystemBackend {
...
                     ),
                     buffer: None,
-                },
-            ));
+                }
+                .into(),
+            )?;
+            return Ok(());
         }
         let file = file.unwrap(); // safe because of the if statement above
```
Comment (ibeckermayer) [inline lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs:538]: https://github.com/gravitational/teleport/pull/34082#discussion_r1380912178

@probakowski 
> I'd rather avoid unwrap

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs (hunk 1 above)
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs]: a59a44ede95a22ba2fa1c848c78668c87cf1d71d

File: lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs (hunk 2 above)
Comment (ibeckermayer) [reply lib/srv/desktop/rdp/rdpclient/src/rdpdr/filesystem.rs]: a59a44ede95a22ba2fa1c848c78668c87cf1d71d
